dotnet/src/SemanticKernel.Abstractions/Filters/AutoFunctionInvocation/IAutoFunctionInvocationFilter.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel;
+
+#pragma warning disable CA1716 // Identifiers should not match keywords (Func<FunctionCallInvocationContext, Task> next)
+
+/// <summary>
+/// Interface for filtering actions during automatic function invocation.
```

Comment:
In manual function calling, user will be responsible for invocation loop. So, user will have to write custom logic that doesn't go through the LLM one last time. In some cases, loop is not even required, user can send request to LLM, execute required functions and return result. But since it's manual function calling, user has more control over it and user decides how the flow should look like.